tests/lib/rules/no-dupe-else-if.js

```diff
@@ -0,0 +1,317 @@
+        },
+        {
+            code: "if (a) {} else if (b) {} else if (c && (a || b)) {}",
+            errors: [{ messageId: "unexpected", type: "LogicalExpression", column: 35 }]
+        },
+        {
+            code: "if (a) {} else if (b && c) {} else if (d && (a || e && c && b)) {}",
+            errors: [{ messageId: "unexpected", type: "LogicalExpression", column: 40 }]
+        },
+        {
+            code: "if (a || b && c) {} else if (b && c && d) {}",
+            errors: [{ messageId: "unexpected", type: "LogicalExpression", column: 30 }]
+        },
+        {
+            code: "if (a || b) {} else if (b && c) {}",
+            errors: [{ messageId: "unexpected", type: "LogicalExpression", column: 25 }]
+        },
+        {
+            code: "if (a) {} else if (b) {} else if ((a || b) && c) {}",
+            errors: [{ messageId: "unexpected", type: "LogicalExpression", column: 35 }]
+        },
+        {
+            code: "if ((a && (b || c)) || d) {} else if ((c || b) && e && a) {}",
+            errors: [{ messageId: "unexpected", type: "LogicalExpression", column: 39 }]
+        },
+        {
+            code: "if (a && b || b && c) {} else if (a && b && c) {}",
+            errors: [{ messageId: "unexpected", type: "LogicalExpression", column: 35 }]
+        },
+        {
+            code: "if (a) {} else if (b && c) {} else if (d && (c && e && b || a)) {}",
+            errors: [{ messageId: "unexpected", type: "LogicalExpression", column: 40 }]
+        },
+        {
+            code: "if (a || (b && (c || d))) {} else if ((d || c) && b) {}",
+            errors: [{ messageId: "unexpected", type: "LogicalExpression", column: 39 }]
+        },
+        {
+            code: "if (a || b) {} else if ((b || a) && c) {}",
+            errors: [{ messageId: "unexpected", type: "LogicalExpression", column: 25 }]
+        },
+        {
+            code: "if (a || b) {} else if (c) {} else if (d) {} else if (b && (a || c)) {}",
+            errors: [{ messageId: "unexpected", type: "LogicalExpression", column: 55 }]
+        },
+        {
+            code: "if (a || b || c) {} else if (a || (b && d) || (c && e)) {}",
+            errors: [{ messageId: "unexpected", type: "LogicalExpression", column: 30 }]
+        },
+        {
+            code: "if (a || (b || c)) {} else if (a || (b && c)) {}",
+            errors: [{ messageId: "unexpected", type: "LogicalExpression", column: 32 }]
+        },
+        {
+            code: "if (a || b) {} else if (c) {} else if (d) {} else if ((a || c) && (b || d)) {}",
+            errors: [{ messageId: "unexpected", type: "LogicalExpression", column: 55 }]
+        },
+        {
+            code: "if (a) {} else if (b) {} else if (c && (a || d && b)) {}",
+            errors: [{ messageId: "unexpected", type: "LogicalExpression", column: 35 }]
+        },
+        {
+            code: "if (a) {} else if (a || a) {}",
```

Comment:
unrelated: is there a rule to warn something like: `a || a;` -- it seems a common mistake.

Reply:
I don't think so. There was an idea to check this in [#12097 (comment)](https://github.com/eslint/eslint/issues/12097#issuecomment-522388503)

Reply:
Off-topic, maybe we should reopen #12097 or open a new issue.